src/dotnet-test-xunit/BinaryWriterTestDiscoverySink.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+using System.IO;
+using Microsoft.Extensions.Testing.Abstractions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Xunit.Runner.DotNet
+{
+    public class BinaryWriterTestDiscoverySink : ITestDiscoverySink
+    {
+        private readonly BinaryWriter _binaryWriter;
+
+        public BinaryWriterTestDiscoverySink(BinaryWriter binaryWriter)
+        {
+            _binaryWriter = binaryWriter;
+        }
+
+        public void SendTestCompleted()
+        {
+            _binaryWriter.Write(JsonConvert.SerializeObject(new Message
+            {
+                MessageType = "TestRunner.TestCompleted"
```

Comment:
is this sufficient? I thought we needed some id information for the test so we know which one completed...